### DISCUSSION TITLE
Eastron SD72M energymeters

### DISCUSSION THREAD

**Discussion by VelocityMaximus** [serkri/SmartEVSE-3#187] [Q&A]
I have 2 SmartEVSE-3 chargers and 2 Eastron SDM72D-M energymeters. The energymeters are now connected via modbus on HomeAssistant.
These meters do not measure current and voltage but only import and export kW and kWh. I tested the meters and their data show up when selecting Eastron 3P.

Is it OK to connect them directly to the EVSE module and read them there or are there any possible issues that current and voltage are not read and stay zero?

**Comment by dingo35** [comment]
You are wrong, I have been using SDM72D-M meters for over a year now and they measure all kinds of parameters, current, voltage, frequency, power factor, all per phase, everything you want. Its perfectly safe to connect them to your SmartEVSE's modbus.

**Reply by VelocityMaximus** [reply]
It did not show up the V and A values in the EVSE statuspage when I was testing.
Also per info from this site: https://webshop.cedel.nl/SDM72D-Modbus-MID  There is a note: 'Meet geen A, F, PF en kVa en meet niet de individuele fasen.'

The registers as per original manual:
30053 Total system power. 4 Float W 00 34
30073 Total Import kWh 4 Float kWh 00 48
30075 Total Export kWh. 4 Float kWh 00 4A
30343 Total kWh (1) 4 Float kWh 01 56
30385 resettable total active energy (1) 4 Float kWh 01 80
30389 resettable import active energy 4 Float kWh 01 84
30391 resettable export active energy 4 Float kWh 01 86
31281 Total import active power 4 Float W 05 00
31283 Total export active power 4 Float W 05 0

**Comment by dingo35** [comment]
Ok  I have exactly the same model here, but it says SW v1.2 2020 while on the cedel website it says SW v1.2 2018.
So it seems we have a manufacturer that changes specs without change model/type, and a webshop that sells old models at top prizes (you can get the same model 2023 Eastron for a little over half the price).

So if it really doesnt support those registers on modbus, the SmartEVSE would timeout on those because it really needs those current values when in Smart or Solar mode.
So that deems them unusable for Smart/Solar use, and for Normal mode you dont need a kWh meter...

**Reply by VelocityMaximus** [reply]
If I turn on the meters it shows 02 01.02 as version. Can you check what yours is displaying?

**Comment by dingo35** [comment]
02.01 03

**Comment by VelocityMaximus** [comment]
The firmware version is older and does not provide all needed values for smart and solar charging.

**Reply by mstegen** [reply]
I also have this meter, and it reports 02.01.02. at powerup.
It definitely works correctly with the SmartEVSE. 

It's however important to connect all phases, otherwise it's not guaranteed that the measured values are correct.
If you are testing on one phase, you should connect L1, L2 and L3 together.

**Reply by dingo35** [reply]
Just to be complete, you _can_ use it as EVmeter (because then only energy is used), but you _cannot_ use it for MainsMeter....

EDIT: And with the serkri firmware your modbus will endure a lot of timeouts, since also for EV currents are read....

**Reply by VelocityMaximus** [reply]
I have mailed Eastron about updating the firmware but have no response yet.
I will test the meters asap (when there is no rain) on the EVSE modules and see if all the values are presented with the 02.01.02.
Last time I tested I could read the power and usage but did not see current and voltage then.
While testing I had to connect all phases on the testbench, otherwise the meter was in a restart loop and could not change the address.

**Comment by dingo35** [comment]
Yes but you are right, firmware version 020102 only has some energy and power parameters, voltage and currents are introduced in 020103.
